serverless/functions/reference-content/configure-trigger-inputs.mdx

```diff
@@ -0,0 +1,60 @@
+---
+meta:
+  title: Considerations to configure event retention for trigger inputs
+  description: This page explains how to configure event retention for trigger inputs
+content:
+  h1: Considerations to configure event retention for trigger inputs
+  paragraph: This page explains how to configure event retention for trigger inputs
+tags: functions
+dates:
+  validation: 2023-06-09
+  posted: 2023-06-09
+categories:
+  - serverless
+  - functions
+---
+
+<Message type="important">
+  This feature is currently in beta and only allows a max of 10 inflight requests, so functions that are configured to have more than 10 replicas won't be used at full. In the future this limit will be removed. Please, take that into consideration while using the following formulas.
+</Message>
+
+Triggers get events from an input, such as a SQS queue, and forward them to a function, which will scale up according to its settings to accommodate the workload. This process uses back pressure, so that no new events are read if the function instances have not yet finished processing the previous ones. It is important to note that triggers only keep a buffer of the messages that are inflight, and they DO NOT drain all the messages of the input in advance.
+
+As a result, in some scenarios such as event bursts or slow computations, events may stay in the input buffer for a while before being consumed. If there is some kind of message-level timeout in the input, as in the case of [MessageRetentionPeriod](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html) in SQS queues, this may lead to events being deleted before triggering the function.
+
+The implementation of the core trigger behavior strives to be input agnostic, so it is the responsibility of users to configure the input buffers according to their use case to avoid losing events. Also, some users may be okay with old messages being automatically deleted if they haven't been processed in time.
```

Comment:
"... core trigger behavior is input agnostic, it is therefore your responsibility to configure the input buffers according to your use case..."